STYLE.md

```diff
@@ -131,15 +131,188 @@ if status is not pebble.ServiceStatus.ACTIVE:
-### Spell out abbreviations
+Abbreviations and acronyms in docstrings should usually be spelled out:
 
-Abbreviations and acronyms in docstrings should usually be spelled out, for example, "for example" rather than "e.g.", "that is" rather than "i.e.", "and so on" rather than "etc", and "unit testing" rather than UT.
+- "for example" rather than "e.g."
+- "that is" rather than "i.e."
+- "and so on" rather than "etc"
+- "unit testing" rather than UT, and so on
 
 However, it's okay to use acronyms that are very well known in our domain, like HTTP or JSON or RPC.
+
+#### Use sentence case in headings
+
+`## Use sentence case for headings`, instead of `## Use Title Case for Headings`.
+
+#### Be consistent
+
+Be consistent in every possible way as much as possible, and stay context-consistent.
+
+Choice of words: For example, if the whole document uses "mandatory", you probably shouldn't use "required" in a newly added paragraph. For another example, if the whole doc uses "list foo" when adding new content, don't use "get a list of bar".
+
+Headings/lists: For example, if a doc is called "How to use Pebble", with sections "API", "CLI commands", "Health checks", and "Specification", use the heading "Environment variables" for a new section about environment variables, instead of the heading "Pebble environment variables".
+
+#### Be precise
+
+Be precise in names and verbs. Use precise verbs to describe the behaviour. For example, the appropriate description of `/v1/services` is "list services", while "get a service" is probably a better fit for `/v1/services/{name}`.
+
+Split distinct ideas: Use separate sentences/clauses and avoid cramming. Don’t merge unrelated details (e.g., parameter format + default behaviour) into a single phrase.
+
+Example:
+
+- Avoid: "The names of the services to get, a comma-separated string. If empty, get all services."
+- Prefer: "The names of the services to get. Specify multiple times for multiple values. If omitted, the method returns all services." (In three short precise sentences we've covered what the parameter does, usage details, and behavioral details.)
+
+Example:
+
+- Avoid: "For reference information about the API, see [API and clients] and [API]." (What's the difference of these two?)
+- Prefer: "For an explanation of API access levels, see [API and clients]. For the full API reference, see [API reference]." (Clear)
+
+#### Don't over-promote
+
+Avoid overstatement: Only use adjectives like "comprehensive", "powerful", or "robust" when the feature truly meets the description.
+
+Example:
+
+- Avoid: "Pebble provides a comprehensive health check feature"
+- Prefer: "Pebble provides a health check feature"
+
+Let features speak for themselves, describe actual capabilities and use measurable terms when possible (e.g., "supports 3 types of checks" instead of "versatile checking")
+
+Future-proofing:
+
+- Avoid absolute terms unless permanently true
+- Qualify with "currently" when describing evolving features
```

Comment:
I actually think we should drop this. It's usually just okay to omit "currently" and have the current docs reflect the current state. Otherwise if we were consistent there'd be "currently" everywhere.

Alternatively we could give a specific example of when this would apply.